kore/test/Test/Kore/Step/Axiom/Matcher.hs

```diff
@@ -1029,6 +1043,816 @@ test_unificationWithAppMatchOnTop =
         assertEqualWithExplanation "" expected actual
     ]
 
+test_matchingBuiltins :: [TestTree]
+test_matchingBuiltins =
+    Foldable.fold
+        [ matchingBool
+        , matchingInt
+        , matchingString
+        , matchingList
+        , matchingSet
+        , matchingMap
+        ]
+
+matchingBool :: [TestTree]
```

Comment:
Could we make this a `TestTree` using `testGroup` so that `test_matchingBuiltins` above isn't flat? I think it will make the test output clearer.